crates/oxc_cli/src/lib.rs

```diff
@@ -36,9 +54,13 @@ impl Cli {
             ret.errors
         };
 
-        for diagnostic in diagnostics {
-            let diagnostic = diagnostic.with_source_code(source_text.clone());
-            println!("{diagnostic:?}");
-        }
+        diagnostics
+            .into_iter()
+            .map(|diagnostic| {
+                let diagnostic = diagnostic.with_source_code(source_text.clone());
+                println!("{diagnostic:?}");
+                diagnostic
+            })
+            .collect()
```

Comment:
Collecting into an array allocates memory, so strive to allocate less in Rust.

so in this case you can probably do a sum and then return total count and print them inside the `report` method. Something like "Found 123 diagnostics."

Reply:
I'll do this after your PR.